### ISSUE TITLE
TImestamp not included in searchTx() response

### ISSUE THREAD

**Issue by Goochie** [cosmos/cosmjs#1181]
Is there any reason that the timestamp is not included in the returned response when it is parsed in the following function ..

private async txsQuery(query: string): Promise<readonly IndexedTx[]> {
    const results = await this.forceGetTmClient().txSearchAll({ query: query });
    return results.txs.map((tx) => {
      return {
        height: tx.height,
        hash: toHex(tx.hash).toUpperCase(),
        code: tx.result.code,
        rawLog: tx.result.log || "",
        tx: tx.tx,
        gasUsed: tx.result.gasUsed,
        gasWanted: tx.result.gasWanted,
      };
    });
  }

.......... as its not included what is the expectations with regards to getting timestamp, do we have to call getBlock() ?

**Comment by Goochie** [comment]
I have decoded the transaction and that doesn't hold a timestamp ... what am i missing


decodeTxRawFunc(transactions[0].tx)

**Comment by webmaster128** [comment]
Transactions don't have a timestamp. The only time that exists is the timestamp of the block in which the transaction was executed. To get that you need to take the height and query the block information at that height.

**Comment by Goochie** [comment]
Thanks again :) ...... the only thing i would add is that when using the REST api you get all transactions and the timestamp ... this is where a lot of my confusion stems ie i was using the REST end points to test out data fetch ie /cosmos/tx/v1beta1/txs .... then moved to cosmJS and nothing made sense but that is all part of on boarding yourself to new tech ...... once again thanks.

**Comment by webmaster128** [comment]
Yeah, right, makes sense. The reason is that the "REST" API server does exactly what I described behind the scenes to provide a convenient response to users. It's always the block time of the block in which the transaction is executed.

**Comment by webmaster128** [comment]
Closing as resolved. Please feel free to re-open if you have follow-up questions that fall under the same title.
